treestats_paper.tex

```diff
@@ -1266,13 +1267,33 @@ \subsection*{Algorithms}
 %%%%%%%%%%%%%%%%%%%%%
 \subsection*{Speed}
 
-compare speed to scikit-allel
+\krt{need to sort out the big-O issue here}
+We used coalescent simulations to compare the performance of calculating Tajima's \citeyearpar{Tajima1989-de} $D$
+statistic from tree sequences to calculating it from integer matrices containing genotypes at all variable positions.
+\autoref{fig:stats_performance} shows that the tree sequence methods implemented in \texttt{tskit} process variant data
+three orders of magnitude faster than matrix-based approaches once the sample size is above $n \approx 500$ haploids.
+The expected number of mutations for each replicate is $10^4 \times \sum_{i=1}^{n-1}\frac{1}{i}$, and thus ranges from
+$\approx 32,000$ to $\approx 138,000$ \citep{Watterson1975-ej}.
+\autoref{fig:stats_performance} only considers the time required to calculate the statistic. It does not include the time
+required to build the genotype matrix, nor the memory required to store it.  For the largest samples size of $n=10^6$,
+the matrix is approximately 200 gigabytes, and thus not practical on many systems.
```

Comment:
Perhaps note how much memory the tree sequence uses?

Reply:
This is not easily knowable.  The tree sequence objects don't have a member function to return their allocated size.  The size of the genotype matrix can be obtained directly from the sample size + number of mutations.  

One option would be to rerun the benchmarks, and output the size of node + edge tables.  Then, we know that there are 5 or 6 arrays making up the linked list for a tree, and each is `len(node table)`.  From that, I can get the size pretty much right.  Thoughts?

Reply:
Well, we don't need to be totally precise here. Maybe it's enough to either (a) open python, and the tree sequence in it, and see how much memory is used, or (b) just say "the tree sequence uses much less memory"?

Reply:
The exact tree sequences are not stored, but I could fudge it by simulating one real quick...

Reply:
Done